helix-term/src/application.rs

```diff
@@ -393,38 +393,51 @@ impl Application {
         self.editor.refresh_config();
     }
 
+    /// refresh language config after config change
+    fn refresh_language_config(&mut self) -> Result<(), Error> {
+        let syntax_config = helix_core::config::user_syntax_loader()
+            .map_err(|err| anyhow::anyhow!("Failed to load language config: {}", err))?;
+
+        self.syn_loader = std::sync::Arc::new(syntax::Loader::new(syntax_config));
+        for document in self.editor.documents.values_mut() {
+            document.detect_language(self.syn_loader.clone());
+        }
+
+        Ok(())
+    }
+
     /// Refresh theme after config change
-    fn refresh_theme(&mut self, config: &Config) {
+    fn refresh_theme(&mut self, config: &Config) -> Result<(), Error> {
         if let Some(theme) = config.theme.clone() {
             let true_color = self.true_color();
-            match self.theme_loader.load(&theme) {
-                Ok(theme) => {
-                    if true_color || theme.is_16_color() {
-                        self.editor.set_theme(theme);
-                    } else {
-                        self.editor
-                            .set_error("theme requires truecolor support, which is not available");
-                    }
-                }
-                Err(err) => {
-                    let err_string = format!("failed to load theme `{}` - {}", theme, err);
-                    self.editor.set_error(err_string);
-                }
+            let theme = self
+                .theme_loader
+                .load(&theme)
+                .map_err(|err| anyhow::anyhow!("Failed to load theme `{}`: {}", theme, err))?;
+
+            if true_color || theme.is_16_color() {
+                self.editor.set_theme(theme);
+            } else {
+                return Err(anyhow::anyhow!(
+                    "theme requires truecolor support, which is not available"
+                ));
```

Comment:
Very minor thing but this can be written with `anyhow::bail!`:

```rs
anyhow::bail!("theme requires truecolor support, which is not available")
```